### DISCUSSION TITLE
Can't compile output example

### DISCUSSION THREAD

**Discussion by greno** [pschatzmann/arduino-audiokit#61] [Q&A] [answered]
I have an AI Thinker Audio Kit (listed on Amazon as JESSINIE ESP32 Aduio Kit ESP32 WiFi Bluetooth Module ESP32-A1S Module Audio Development Board BLE Low Power Dual-core 64Mb Serial Adapter Port to WiFi Board).  As per the FAQ, I selected the ESP32 Dev Module as the board.  Using Arduino 2.0.3.

I'm getting the following compiler errors when trying examples -> audio kit -> output:

In file included from C:\Users\gregg\AppData\Local\Temp\.arduinoIDE-unsaved2023011-14748-15955me.jjnwk\output\output.ino:10:
c:\Users\gregg\OneDrive\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h: In member function 'i2s_config_t AudioKitConfig::i2sConfig()':
c:\Users\gregg\OneDrive\Documents\Arduino\libraries\arduino-audiokit-main\src/AudioKitHAL.h:129:34: warning: narrowing conversion of '((AudioKitConfig*)this)->AudioKitConfig::sampleRate()' from 'uint32_t' {aka 'unsigned int'} to 'int' inside { } [-Wnarrowing]
         .sample_rate = sampleRate(),
                        ~~~~~~~~~~^~
c:\Users\gregg\OneDrive\Documents\Arduino\libraries\arduino-audiokit-main\src\audio_board\ai_thinker_es8388_3478\board_pins_config.c: In function 'get_i2s_pins':
c:\Users\gregg\OneDrive\Documents\Arduino\libraries\arduino-audiokit-main\src\audio_board\ai_thinker_es8388_3478\board_pins_config.c:67:21: error: 'i2s_pin_config_t' {aka 'struct <anonymous>'} has no member named 'mck_io_num'; did you mean 'bck_io_num'?
         i2s_config->mck_io_num = GPIO_NUM_0;
                     ^~~~~~~~~~
                     bck_io_num

I can see the i2s_pin_config_t struct is defined in arduino-audiokit-main\src\audio_hal\audio_gpio.h, but as the error says, it does not define the mck_io_num.

Any suggestions?  
Thanks,
-Gregg

**Comment by pschatzmann** [comment]
As you can see these are just warnings. Just make sure that you don't treat warnings as errors!

**Comment by greno** [comment]
Thanks for the quick reply, but that last one is an error and not a warning.  The last couple of lines that I should have included:

exit status 1

Compilation error: exit status 1

**Comment by pschatzmann** [comment, marked as answer]
Make sure that you use an actual Arduino ESP32 version: The latest version is 2.0.6!

**Reply by greno** [reply]
That was it! I was using the wrong board URL.  After installing 2.0.06, it compiles and I'm getting a tone.  Thanks!

**Reply by pschatzmann** [reply]
What version was giving this error ? I am currently checking for ESP_IDF_VERSION_MAJOR >= 4   to determine if the mck_io_num pin is available

**Reply by greno** [reply]
I believe it was 2.0.0

**Reply by pschatzmann** [reply]
I committed some corrections to make sure that 2.0.0 is compiling w/o errors as well
